openedx/core/djangoapps/enrollments/views.py

```diff
@@ -987,3 +994,164 @@ def get_queryset(self):
+
+        if not all_users:
+            course_enrollments = course_enrollments.filter(user__username=username).order_by('created')
+
+        courses = {}
+        for course_enrollment in course_enrollments:
+            try:
+                course_list = courses.get(course_enrollment.course_id)
+                if course_list:
+                    course, course_children = course_list
+                else:
+                    course = get_course(course_enrollment.course_id, depth=4)
+                    course_children = course.get_children()
+                    courses[course_enrollment.course_id] = [course, course_children]
+            except ValueError:
+                continue
+            course_data = self._get_course_data(course_enrollment, course, course_children)
+            data.append(course_data)
+
+        return Response({'results': data})
+
+    def _get_problem_data(self, course_enrollment, component):
+        """
+        Get problem data from a course enrollment.
+
+        Args:
+        -----
+        course_enrollment: Course Enrollment.
+        component: Component to analyze.
+        """
+        problem_data = {
+            'location': str(component.location),
+            'name': component.display_name,
+            'submission_history': [],
+            'data': component.data
+        }
+
+        csm = StudentModule.objects.filter(
+            module_state_key=component.location,
+            student__username=course_enrollment.user.username,
+            course_id=course_enrollment.course_id)
+
+        scores = BaseStudentModuleHistory.get_history(csm)
+        for i, score in enumerate(scores):
+            if i % 2 == 1:
+                continue
+
+            state = score.state
+            if state is not None:
+                state = json.loads(state)
+
+            history_data = {
+                'state': state,
+                'grade': score.grade,
+                'max_grade': score.max_grade
+            }
+            problem_data['submission_history'].append(history_data)
+
+        return problem_data
+
+    def _get_course_data(self, course_enrollment, course, course_children):
+        """
+        Get course data.
+
+        Params:
+        --------
+
+        course_enrollment (CourseEnrollment): course enrollment
+        course: course
+        course_children: course children
+        """
+
+        course_data = {
+            'course_id': str(course_enrollment.course_id),
+            'course_name': course.display_name_with_default,
+            'user': course_enrollment.user.username,
+            'problems': []
+        }
+        for section in course_children:
+            for subsection in section.get_children():
+                for vertical in subsection.get_children():
+                    for component in vertical.get_children():
+                        if component.location.category == 'problem' and getattr(component, 'has_score', False):
```

Comment:
```suggestion
                        if block.category == 'problem' and getattr(block, 'has_score', False):
```

This should be sufficient (though I didn't test it).